### ISSUE TITLE
Fires layers - popup issue

### ISSUE THREAD

**Issue by csphang** [wri/gfw-mapbuilder#406]
Pop-ups only work for fires from the past 24 hours. 

If you switch to another time period and click on a fire point which older than the past 24 hours the pop-up does not show up. I am attaching a video in which I switch the time period to the past week and once I do this some of the pop-ups stop working. 
[firepopups.zip](https://github.com/wri/gfw-mapbuilder/files/3273906/firepopups.zip)

**Comment by KaylaKremer** [comment]
@SwampGuzzler Did you already fix this?

**Comment by SwampGuzzler** [comment]
@KaylaKremer No! This is a new issue!

**Comment by richardbarad** [comment]
@SwampGuzzler & @KaylaKremer,

I tested this in this build: http://alpha.blueraster.io/gfw-mapbuilder/406-fires-bug/, which I think is 

It looks like the pop-ups are working when you select past 24 hours or past 48 hours, but not when you select past 72 hours or past week. I see the same behavior on both VIIRS and MODIS. Please see attached video. 

[firesbug.zip](https://github.com/wri/gfw-mapbuilder/files/3526527/firesbug.zip)

**Comment by KaylaKremer** [comment]
@richardbarad I think I haven't updated that build for a while, but I believe we solved this issue. We had to update the VIIRS 1 YR fire service (We use this service for anything beyond 72 hours). We had only updated the service on staging, so we just need to update it on production. I can provide an updated build once this is done.

**Comment by KaylaKremer** [comment]
@richardbarad I've gone ahead and updated the build just now for you to test again and be sure. It's still using our 1 YR VIIRS service on staging, which we plan to switch out for the production one:

http://alpha.blueraster.io/gfw-mapbuilder/406-fires-bug/

**Comment by richardbarad** [comment]
@KaylaKremer. I just checked this on MODIS and it looks there is still an issue. If I:

1. Turn on the MODIS Active Fires layer
2. Zoom in on the map to central DRC on an area with a large number of fires
3. Toggle from past 24 hours to 48 hours, and identify a point which was not present in the 24 hours layer
4. Click on that point
5. The point does not show any results in the attribute table

I noticed that it looks like the query is always being made on the `https://gis-gfw.wri.org/arcgis/rest/services/Fires/FIRMS_Global_MODIS_24hrs/MapServer/21` service even when you click on a point in the 48 hours service. See screenshot below and the attached video. 

[fires_popupbug.zip](https://github.com/wri/gfw-mapbuilder/files/3531440/fires_popupbug.zip)

![pop-Fires](https://user-images.githubusercontent.com/40440743/63539910-175e7080-c4e9-11e9-93ee-065ddd845ad3.png)

**Comment by KaylaKremer** [comment]
@richardbarad , Thanks for the feedback! Just to confirm, you are only having the issue with MODIS, not also VIIRS correct?

**Comment by richardbarad** [comment]
@KaylaKremer. I am seeing it for VIIRS as well, except VIIRS seem to query the 48 hour layer regardless of which layer is turned on. `https://gis-gfw.wri.org/arcgis/rest/services/Fires/FIRMS_Global_VIIRS_48hrs/MapServer/21/query`

**Comment by csphang** [comment]
@richardbarad Could this be a cache issue? I just replicated the workflow and it worked on my PC. Can you try in incognito and see if you still experience the same behavior?

**Comment by csphang** [comment]
@richardbarad we see the issue in your video but are unable to reproduce! I recorded what I'm seeing on my end. Let me know if you want to setup a screenshare for this?

Link to download video: https://chilp.it/2b26437

**Comment by richardbarad** [comment]
@csphang. Sure, we can do a quick screen-share on Monday. 

I wonder if you had clicked on points in the 48 hours layer before starting to create the video in the same view session? That seems to prevent the bug from occurring as the pop-ups always seem to query against the first layer I click on. (i.e.: If I click on a pop-up in the 48 hours layer, then switch to 24 hours, then switch back to 48 hours it will work). Happy to have a screen-share if you still can not replicate.

Richard

**Comment by SwampGuzzler** [comment]
@KaylaKremer Our thought - different layer `id` for each of the Fires Layers (for both VIIRS annnd MODIS) - ex: `VIIRS_ACTIVE_FIRES` --> `VIIRS_ACTIVE_FIRES_24` && `VIIRS_ACTIVE_FIRES_48` && `VIIRS_ACTIVE_FIRES_72` && `VIIRS_ACTIVE_FIRES_7` && `VIIRS_ACTIVE_FIRES_1` which is the ID we give it in the new layer on creation in the `updateFiresLayerDefinitions` function based off of Alt State!!

**Comment by KaylaKremer** [comment]
@richardbarad ,

I think I made good progress with this ticket by myself today. Here's a new build to test out: http://alpha.blueraster.io/gfw-mapbuilder/406-update-fires-popups-progress/

Let me know if the VIIRS and MODIS fires is now working as it should. If not, I will consult with Lucas about it tomorrow when he returns to work.

**Comment by richardbarad** [comment]
@KaylaKremer. I think this bug is still present.

IF)

1. Turn on fires in past 24 hours

2. Click on a point to open up the pop-up

3. Toggle to switch to a different time period, and then click on another point which was not visible in the 24 hours layer

4. The fires pop is not visible

[fires_bug.zip](https://github.com/wri/gfw-mapbuilder/files/3688599/fires_bug.zip)

**Comment by KaylaKremer** [comment]
Hi @richardbarad ,

Here's the latest progress we have for this ticket. I made a new build: http://alpha.blueraster.io/gfw-mapbuilder/406-putting-out-fires/

We have almost everything working. The 1 year service (and potentially 72HRs) doesn't work. We are still working on figuring a way to update the queries appropriately.

**Comment by richardbarad** [comment]
@KaylaKremer. 

I did a review. Please also check the custom range / 1 year layers. 

1. It looks all the fires load for the past year even when the user specifies a custom range for both VIIRS and MODIS.

2. The pop-ups are also not working for the 1 year layers for both VIIRS and MODIS

**Comment by richardbarad** [comment]
@KaylaKremer. One more bug:

1.	Turn on VIIRS
2.	Switch VIIRS to past 48 hours
3.	Turn off VIIRS, the layer does not turn off!

![image](https://user-images.githubusercontent.com/40440743/66517002-4d16e300-eab0-11e9-8b6c-78321925f412.png)

**Comment by KaylaKremer** [comment]
@richardbarad ,

I updated the build again and I think we now have everything working!
http://alpha.blueraster.io/gfw-mapbuilder/406-putting-out-fires/
